agr_literature_service/api/routers/topic_entity_tag_router.py

```diff
@@ -133,3 +136,40 @@ def show_all_reference_tags(curie_or_reference_id: str,
 def get_map_entity_curie_to_name(curie_or_reference_id: str,
                                  db: Session = db_session):
     return topic_entity_tag_crud.get_map_entity_curie_to_name(db, curie_or_reference_id)
+
+
+def revalidate_tags_process_wrapper(already_running, email: str, delete_all_first: bool, db: Session):
+    try:
+        already_running.value = True
+        topic_entity_tag_crud.revalidate_all_tags(db=db, email=email, delete_all_first=delete_all_first)
+    finally:
+        already_running.value = False
+
+
+@router.get('/revalidate_all_tags/',
+            status_code=200)
+def revalidate_all_tags(email: str,
+                        delete_all_tags_first: bool = False,
+                        user: OktaUser = db_user,
+                        db: Session = db_session):
+    if not user.groups or "SuperAdmin" not in user.groups:
```

Comment:
SuperAdmin seems good, but I don't know who's meant to run this.  For example, you and I are, but Shuai and Pinglei aren't.

Reply:
Good to know, it should be ok but I'll ask Ceri at standup.

Reply:
Ceri approved this